### ISSUE TITLE
Bug - Dark mode does not respect system settings and defaults to light mode on every page load

### ISSUE THREAD

**Issue by garrett** [patternfly/patternfly-org#3867]
**Describe the problem**

All pages for PatternFly.org are in light mode by default, regardless of system settings. Manually switching to dark mode does not work, except while on that page.

**Expected behavior**

PatternFly.org should default to the system color scheme. There can be a preference to override the theme (especially to preview widgets in light/dark modes for reference while developing), but it should be the system mode by default and there should be a way to get back to system default too.

**Any other information?**

Sadly, I often have vision issues where I need dark mode, either due to migraine or allergy problems. I have to set my system and apps to dark mode.

When it's not too bad, I still try to work, but blasting bright light in my face is physically painful and will make my headache and/or eye pain worse (depending on the cause). Too much brightness, and I can't even work.

(I also have astigmatism, which means I need light mode too, when I'm able to tolerate it. But when I need it, using dark mode overrides that for me.)

This was why I build dark mode in Cockpit and pushed for it in PatternFly. :wink: We should also have it for the website, especially since PatternFly has the ability to do so.

There should be a trinary setting, with auto, light, and dark. In Cockpit we have a switcher with text in the menu, like this:

![screenshot of Cockpit's light/dark mode switcher](https://github.com/patternfly/patternfly-org/assets/10246/444dcd44-aa30-4bf2-9dc9-f1723d7771be)

GNOME has an icon with text, which has light and dark. It's a toggle, as it is the system so it sets the system color scheme:

![image](https://github.com/patternfly/patternfly-org/assets/10246/e6221b83-6030-4382-b709-74235ad5b5b1)

Discord has a trinary state, of light, dark, and system:

![image](https://github.com/patternfly/patternfly-org/assets/10246/5319ab19-56e1-465c-bb4c-8e612f10c581)

Since PatternFly is a reference site and people may need to see light and dark modes for widgets in cases where they do not want to change their system settings, I think it needs 3 manual states. As it uses a sun and moon icon currently, I'd suggest something like what GNOME uses for an icon, the half filled circle: ◐

This means, it could be accessible by default with the system setting and then someone could override it for reference (when they need reference dark/light widgets).

**Comment by garrett** [comment]
Here's a mockup of how it could look, based on the current site:

![pf-dark-mode](https://github.com/patternfly/patternfly-org/assets/10246/1d808a2f-e0a8-44fd-9077-084d2023530b)

Here is the SVG source:

![pf-dark-mode](https://github.com/patternfly/patternfly-org/assets/10246/15ef84ba-89a2-48ff-8fb1-88da65c085a6)

I used the sun and moon from the website, and made a light/dark icon. The light/dark system icon and the moon icon are the same sizes: 14 px. The sun is 20. The goal was to have each be roughly 16px wide, but the visual weight of the sun versus the moon is quite different, due to the sun's inner part and the rays. To make them all the same visual weight, I nudged down the system and dark mode icons and nudged up the light mode sun.

Here's an optimized (with svgo) pixel-perfect SVG icon I made for the system light/dark icon @ 14px:

![pf-dark-mode-system-paths-opt](https://github.com/patternfly/patternfly-org/assets/10246/d239d781-fa2d-4857-ad5f-4e60ba0696b7)
[pf-dark-mode-system-paths-opt.svg](https://github.com/patternfly/patternfly-org/assets/10246/d239d781-fa2d-4857-ad5f-4e60ba0696b7)

(It may be invisible in light mode. It does show up in dark mode, however. :wink: and the top bar on the PatternFly site is always dark.)

The icons should be obvious by context, but they should _also_ have tooltips (which they appear to not have at the moment).
